### PR TITLE
Release prep v0.35.0

### PR DESCRIPTION
**Pull request by Aneurysm9**
**Description:** Updates `go.mod` for contrib modules and adds `CHANGELOG` entries for `v0.35.0`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,27 @@
...
+- chore(dynatrace): use payload limit from api constants (#5077)
+- Add documentation for filelog's new force_flush_period parameter (#5066)
+- Reuse the gzip reader with a sync.Pool (#5145)
+- Add a trace observer when splunkhecreceiver is used for logs (#5063)
+- Remove usage of deprecated pdata.AttributeValueMapToMap (#5174)
+
+## 🧰 Bug fixes 🧰
+
+- Use staleness markers generated by prometheus, rather than making our own (#5062)
+- \[exporter/datadogexporter] skip NaN and infinite values (#5053)
```
Comment (codeboten) [inline CHANGELOG.md:24]:
```suggestion
- [exporter/datadogexporter] skip NaN and infinite values (#5053)
```

Comment (codeboten) [reply CHANGELOG.md:24]: not sure if this is important, but the format for `exporter/datadogexporter` differs from previous versions.

**Review by codeboten — APPROVED** [review]
🎉

**Comment by tigrannajaryan** [comment]
This has merge conflicts.

**Pull request merged**